python/tests/test_writer.py

```diff
@@ -172,6 +172,24 @@ def test_write_modes(tmp_path: pathlib.Path, sample_data: pa.Table):
     assert DeltaTable(path).to_pyarrow_table() == sample_data
 
 
+def test_append_only(tmp_path: pathlib.Path, sample_data: pa.Table):
```

Comment:
**nitpick:** I would suggest renaming this method to `test_append_only_should_append_only_with_the_overwrite_mode` to improve the test readibility.